streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingJobProgressListener.scala

```diff
@@ -101,6 +101,7 @@ private[streaming] class StreamingJobProgressListener(ssc: StreamingContext)
     runningBatchUIData(batchStarted.batchInfo.batchTime) = BatchUIData(batchStarted.batchInfo)
     waitingBatchUIData.remove(batchStarted.batchInfo.batchTime)
 
+    // TODO. this should be fixed when input stream is not receiver based stream.
```

Comment:
That's a good point. I think it is okay to add it for now. At least under normal operation, when things are keeping up, it is useful to see what is the rate. If it does not keep up, then there are other indicators.